gcloud-java-compute/src/test/java/com/google/gcloud/compute/DeprecationStatusTest.java

```diff
@@ -17,40 +17,138 @@
+          .obsolete(OBSOLETE)
+          .deleted(DELETED)
+          .build();
+  private static final DeprecationStatus<DiskTypeId> DISK_TYPE_STATUS_MILLIS =
+      DeprecationStatus.<DiskTypeId>builder(STATUS)
+          .replacement(DISK_TYPE_ID)
+          .deprecated(DEPRECATED_MILLIS)
+          .obsolete(OBSOLETE_MILLIS)
+          .deleted(DELETED_MILLIS)
+          .build();
   private static final DeprecationStatus<MachineTypeId> MACHINE_TYPE_STATUS =
-      new DeprecationStatus<>(DELETED, DEPRECATED, OBSOLETE, MACHINE_TYPE_ID, STATUS);
+      DeprecationStatus.builder(STATUS, MACHINE_TYPE_ID)
+          .deprecated(DEPRECATED)
+          .obsolete(OBSOLETE)
+          .deleted(DELETED)
+          .build();
 
   @Test
-  public void testConstructor() {
+  public void testBuilder() {
+    compareDeprecationStatus(DISK_TYPE_STATUS, DISK_TYPE_STATUS_MILLIS);
     assertEquals(DELETED, DISK_TYPE_STATUS.deleted());
     assertEquals(DEPRECATED, DISK_TYPE_STATUS.deprecated());
     assertEquals(OBSOLETE, DISK_TYPE_STATUS.obsolete());
     assertEquals(DISK_TYPE_ID, DISK_TYPE_STATUS.replacement());
+    assertEquals(DEPRECATED_MILLIS, DISK_TYPE_STATUS.deprecatedMillis());
+    assertEquals(DELETED_MILLIS, DISK_TYPE_STATUS.deletedMillis());
+    assertEquals(OBSOLETE_MILLIS, DISK_TYPE_STATUS.obsoleteMillis());
+    assertEquals(STATUS, DISK_TYPE_STATUS.status());
+    assertEquals(DELETED, DISK_TYPE_STATUS_MILLIS.deleted());
+    assertEquals(DEPRECATED, DISK_TYPE_STATUS_MILLIS.deprecated());
+    assertEquals(OBSOLETE, DISK_TYPE_STATUS_MILLIS.obsolete());
+    assertEquals(DISK_TYPE_ID, DISK_TYPE_STATUS_MILLIS.replacement());
+    assertEquals(DEPRECATED_MILLIS, DISK_TYPE_STATUS_MILLIS.deprecatedMillis());
+    assertEquals(DELETED_MILLIS, DISK_TYPE_STATUS_MILLIS.deletedMillis());
+    assertEquals(OBSOLETE_MILLIS, DISK_TYPE_STATUS_MILLIS.obsoleteMillis());
     assertEquals(STATUS, DISK_TYPE_STATUS.status());
     assertEquals(DELETED, MACHINE_TYPE_STATUS.deleted());
     assertEquals(DEPRECATED, MACHINE_TYPE_STATUS.deprecated());
     assertEquals(OBSOLETE, MACHINE_TYPE_STATUS.obsolete());
+    assertEquals(DEPRECATED_MILLIS, MACHINE_TYPE_STATUS.deprecatedMillis());
+    assertEquals(DELETED_MILLIS, MACHINE_TYPE_STATUS.deletedMillis());
+    assertEquals(OBSOLETE_MILLIS, MACHINE_TYPE_STATUS.obsoleteMillis());
     assertEquals(MACHINE_TYPE_ID, MACHINE_TYPE_STATUS.replacement());
     assertEquals(STATUS, MACHINE_TYPE_STATUS.status());
   }
 
   @Test
+  public void testGettersIllegalArgument() {
+    DeprecationStatus<MachineTypeId> deprecationStatus =
+        DeprecationStatus.builder(STATUS, MACHINE_TYPE_ID)
+            .deprecated("deprecated")
+            .obsolete("obsolete")
+            .deleted("delete")
+            .build();
+    assertEquals("deprecated", deprecationStatus.deprecated());
+    try {
+      deprecationStatus.deprecatedMillis();
+      fail("Expected IllegalArgumentException");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [In JUnit4, use the @Test(expected) annotation to denote tests that should throw exceptions](https://www.codacy.com/app/mziccard/gcloud-java/file/2323737991/issues/source?bid=3140968&fileBranchId=3200309#l103)